File: sequences/infinite-binary-fibonacci.js
var goldenRatio = (1 + Math.pow(5, .5)) / 2;

function InfiniteBinaryFibonacci(start) {
	this.index = (start || 0) + 1;
}

InfiniteBinaryFibonacci.prototype.next = function () {
	var result = getElement(this.index);
	
	this.index++;
	
	return result;
};

function getElement(index) {
	return 2 + Math.floor(index * goldenRatio) - Math.floor((index + 1) * goldenRatio);
}

module.exports = InfiniteBinaryFibonacci;

File: sequences/infinite_binary_fibonacci.py
import math

goldenRatio = (1 + math.pow(5, .5)) / 2;

class InfiniteBinaryFibonacci:
    def __init__(self, **kwargs):
        start = kwargs.get('start', 0)

    	self.index = start + 1

    def next(self):
        result = self.getElement(self.index)
        self.index += 1

        return result

    def getElement(self, index):
	    return int(2 + math.floor(index * goldenRatio) - math.floor((index + 1) * goldenRatio))
